### PR TITLE
Export YAML

### PR DESCRIPTION
**Pull request by sukantoraymond**
To upload cluster YAML file into remote (monitoring & loadtest) host: 
`./bin/avalanche node loadtest CLUSTERNAME SUBNETNAME`

cluster YAML file will be uploaded into remote host, in path `/home/ubuntu/clusterInfo.yaml`

Sample YAML File:

``````
API:
    - CLOUD_ID: i-0c865e75750d157d5
      IP: 3.85.36.201
      REGION: us-east-1
    - CLOUD_ID: i-0415dabcd7e117ba9
      IP: 18.216.76.115
      REGION: us-east-2
VALIDATORS:
    - CLOUD_ID: i-0cd766249a1b270fe
      NODE_ID: NodeID-JFHxwBHPKsUQqfBJdDD2yvvLE8TzqgAES
      IP: 107.22.153.97
      REGION: us-east-1
    - CLOUD_ID: i-0f788025abd6da5fa
      NODE_ID: NodeID-9y3jGFiPLrTgeVfQH16zkL9r55bnx8VYk
      IP: 54.204.68.146
      REGION: us-east-1
    - CLOUD_ID: i-07b1c0a211c190cc9
      NODE_ID: NodeID-PqQLv9DFnN3ogVFHMXHiEncEsDrL1TLKp
      IP: 3.16.113.236
      REGION: us-east-2
    - CLOUD_ID: i-097459bcc24c7d2ce
      NODE_ID: NodeID-PRPvFQf4QAE11yKG6d8VFhAvXZL5ZFMJx
      IP: 18.219.197.235
      REGION: us-east-2
MONITORING:
    CLOUD_ID: i-0ee85dd51d583c8d6
    IP: 18.205.163.47
    REGION: us-east-1
CHAIN_ID: 241Y4mWbveQ966DbjfcxXaySFZ9wYt42i71U1rq1mStenjvfCR
SUBNET_ID: giY8tswWgZmcAWzPkoNrmjjrykited7GJ9799SsFzTiq5a1ML
``````

### CODE REVIEW

File: cmd/nodecmd/loadtest.go [hunk 1]
```diff
@@ -264,9 +284,108 @@ func createLoadTest(_ *cobra.Command, args []string) error {
...
+		if clusterConf.IsAvalancheGoHost(cloudID) {
+			nodeID, err := getNodeID(app.GetNodeInstanceDirPath(cloudID))
+			if err != nil {
+				return err
+			}
+			nodeIDStr = nodeID.String()
+		}
+		roles := clusterConf.GetHostRoles(nodeConfig)
+		switch roles[0] {
+		case "Node":
```
Comment (patrick-ogrady) [inline cmd/nodecmd/loadtest.go:365]: I think we may want to change all instances of the `Node` role to `Validator`

File: cmd/nodecmd/loadtest.go [hunk 2]
```diff
@@ -264,9 +284,108 @@ func createLoadTest(_ *cobra.Command, args []string) error {
...
+		}
+	}
+	clusterConf := clustersConfig.Clusters[clusterName]
+	if err := checkCluster(clusterName); err != nil {
+		return err
+	}
+	validatorCount := 0
+	apiNodeCount := 0
+	for _, cloudID := range clusterConf.GetCloudIDs() {
+		validatorIPEnvPrefix := "IP_VALIDATOR_"
```
Comment (patrick-ogrady) [inline cmd/nodecmd/loadtest.go:346]: We may want to properly nest these fields like:
```yaml
apis:
  - name: XXX
    ip: YYY
  - name: ZZZ
    ip: MMM
```

File: cmd/nodecmd/loadtest.go (hunk 1 above)
Comment (patrick-ogrady) [reply cmd/nodecmd/loadtest.go:365]: (Much better to do in a separate PR)

File: cmd/nodecmd/loadtest.go (hunk 2 above)
Comment (patrick-ogrady) [reply cmd/nodecmd/loadtest.go:346]: Turns out Golang doesn't work well with the current format on the parser side (requires unmarshaling to a struct): https://pkg.go.dev/gopkg.in/yaml.v2@v2.4.0

Comment (patrick-ogrady) [reply cmd/nodecmd/loadtest.go:346]: in the meantime, will unmarshal to: `obj := make(map[string]interface{})`

**Comment by patrick-ogrady** [comment]
I ran this branch and didn't see the `clusterInfo.yaml` file saved to `~/.avalanche-cli/nodes/inventories/CLUSTERNAME/clusterInfo.yaml`

**Comment by patrick-ogrady** [comment]
image

One of these is plural and one isn't

File: cmd/nodecmd/loadtest.go [hunk 3]
```diff
@@ -255,6 +274,19 @@ func createLoadTest(_ *cobra.Command, args []string) error {
 		}
 		ux.Logger.PrintToUser("Separate instance %s provisioned successfully", separateHosts[0].NodeID)
 	}
+
+	subnetID, chainID, err := getDeployedSubnetInfo(subnetName)
```
Comment (arturrez) [inline cmd/nodecmd/loadtest.go:278]: should we check that  entered `subnetName` is correct before using it ?

File: cmd/nodecmd/loadtest.go [hunk 4]
```diff
@@ -263,10 +295,115 @@ func createLoadTest(_ *cobra.Command, args []string) error {
...
+				return model.SubnetID.String(), model.BlockchainID.String(), nil
+			}
+		}
+	}
+	return "", "", fmt.Errorf("unable to find deployed Devnet info at cluster_config.json")
+}
+
+func createClusterYAMLFile(clusterName, subnetID, chainID string, separateHost *models.Host) error {
+	clusterYAMLFilePath := filepath.Join(app.GetAnsibleInventoryDirPath(clusterName), constants.ClusterYAMLFileName)
+	if _, err := os.Stat(clusterYAMLFilePath); err == nil {
```
Comment (arturrez) [inline cmd/nodecmd/loadtest.go:329]: nit: use `utils.FileExists`

File: cmd/nodecmd/loadtest.go [hunk 5]
```diff
@@ -263,10 +295,115 @@ func createLoadTest(_ *cobra.Command, args []string) error {
...
+		nodeIDStr := ""
+		if clusterConf.IsAvalancheGoHost(cloudID) {
+			nodeID, err := getNodeID(app.GetNodeInstanceDirPath(cloudID))
+			if err != nil {
+				return err
+			}
+			nodeIDStr = nodeID.String()
+		}
+		roles := clusterConf.GetHostRoles(nodeConfig)
+		switch roles[0] {
```
Comment (arturrez) [inline cmd/nodecmd/loadtest.go:369]: check if len(roles)>0

File: cmd/nodecmd/wiz.go [hunk 6]
```diff
@@ -247,9 +249,32 @@ func wiz(cmd *cobra.Command, args []string) error {
 	} else {
 		ux.Logger.PrintToUser(logging.Green.Wrap("Devnet %s is successfully created and is now validating subnet %s!"), clusterName, subnetName)
 	}
+
+	if err = deployClusterYAMLFile(clusterName, subnetName); err != nil {
+		return err
+	}
+	ux.Logger.PrintToUser("Successfully deployed cluster information YAML file to external host!")
```
Comment (arturrez) [inline cmd/nodecmd/wiz.go:256]: imho better wording would be to replace 2 lines
ux.Logger.GreenCheckMarkToUser("Cluster info YAML file  /home/ubuntu/clusterInfo.yaml available for loadtest script")

File: cmd/nodecmd/wiz.go [hunk 7]
```diff
@@ -247,9 +249,32 @@ func wiz(cmd *cobra.Command, args []string) error {
...
+	ux.Logger.PrintToUser("Cluster information YAML file can be found at /home/ubuntu/clusterInfo.yaml at external host")
 	return nil
 }
 
+func deployClusterYAMLFile(clusterName, subnetName string) error {
+	separateHostInventoryPath := filepath.Join(app.GetAnsibleInventoryDirPath(clusterName), constants.MonitoringDir)
```
Comment (arturrez) [inline cmd/nodecmd/wiz.go:262]: we use it so many times now it should be abstracted to `func (app *Avalanche)`

File: cmd/nodecmd/wiz.go [hunk 8]
```diff
@@ -247,9 +249,32 @@ func wiz(cmd *cobra.Command, args []string) error {
...
+		return err
+	}
+	subnetID, chainID, err := getDeployedSubnetInfo(subnetName)
+	if err != nil {
+		return err
+	}
+	if err := createClusterYAMLFile(clusterName, subnetID, chainID, separateHosts[0]); err != nil {
+		return err
+	}
+	ux.Logger.PrintToUser("Deploying cluster information YAML file to external host %s", separateHosts[0].GetCloudID())
```
Comment (arturrez) [inline cmd/nodecmd/wiz.go:274]: nit: no need to show this as this will be superfast and we will just report result of it

**Review by patrick-ogrady — APPROVED** [review]

**Pull request merged**